subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -4247,6 +4250,17 @@ static int32_t retransmit_request(struct lwm2m_ctx *client_ctx,
 	return next_retransmission;
 }
 
+static void notify_message_timeout_cb(struct lwm2m_message *msg)
+{
+	if (ext_notify_timeout_cb != NULL) {
+		ext_notify_timeout_cb(msg);
+	}
+#if defined(CONFIG_LWM2M_ENGINE_TIMEOUT_REG_UPDATE)
+	engine_trigger_update(false);
```

Comment:
I just wonder, this mechanism isn't covered by the spec in any way I believe? Perhaps, instead of doing this automatically, we should allow the application to deal with the notification updates on its own? So instead calling app callback and sending the trigger for update automatically, provide a new API function, something like `lwm2m_rd_client_update()`, to trigger the update in the LwM2M state machine?

Reply:
Although this isn't part of the lwm2m specification, I think it would be a useful KConfig option (which is disabled by default), as it fits well with what is proposed for https://github.com/zephyrproject-rtos/zephyr/issues/32610
I can however do as you suggest, and leave it to the application to trigger the update

Reply:
I think that leaving the decision to the application what to do in case of notification timeout gives more flexibility for the user. We see that one solution is to trigger the Register Update, so that the engine can detect connection issues on its own, but some may want to act more actively and restart the registration process right away. 

Generally, I think that within LwM2M engine we should stick as close to the specification as possible - if some non-standard action can be easily achieved from the application (as it is in this scenario), I strongly opt for such a solution.